app/assets/javascripts/shared/editor_toolbar.js

```diff
@@ -15,18 +15,28 @@ class EditorToolbar {
     if (!$target.is("textarea")) { console.log("Can't initialize a rich toolbar on anything but a textarea"); return; }
 
     this.$target = $target;
-    this.opts = { 'include': $target.data('rich-toolbar').split(' ') };
+    this.opts = {
+      'include': $target.data('rich-toolbar').split(' '),
+      'uploader': $target.data('rich-toolbar-uploader')
+    };
+
+    if (this.opts.include.includes('image') && this.opts.uploader === undefined) { console.log("You initialized a RichToolbar with the image uploader option but have not provided an existing uploader to utilize"); return; }
+
     this.affixes = this.affixesLibrary();
     this.init();
   }
 
   init() {
     this.$target.wrap('<div class="editor-field" data-behavior="editor-field"><div class="textarea-container"></div></div>');
-    this.$editorField = this.$target.parents('[data-behavior=editor-field]');
-    this.$editorField.prepend('<div class="editor-toolbar" data-behavior="editor-toolbar"></div>');
-    this.$editorToolbar = this.$editorField.find('[data-behavior=editor-toolbar]');
+    this.$editorField = this.$target.parents('[data-behavior~=editor-field]');
+    this.$editorField.prepend('<div class="editor-toolbar" data-behavior="editor-toolbar" data-edit="#'+ this.$target.data('id') + '"></div>');
+    this.$editorToolbar = this.$editorField.find('[data-behavior~=editor-toolbar]');
 
     this.$editorToolbar.append(this.textareaElements(this.opts.include));
+    this.$fileField = $('<input type="file" name="editor-toolbar-' + Math.random().toString(36) + '[]" multiple style="display: none">');
+    this.$editorToolbar.append(this.$fileField);
```

Comment:
And also, possible to add a check to append the fileField only if there's an `image` listed in rich-toolbar? Now it's always prepending the fileField even if the editor doesn't have an image uploader.

Reply:
Good suggestion.

Reply:
> And also, possible to add a check to append the fileField only if there's an image listed in rich-toolbar? Now it's always prepending the fileField even if the editor doesn't have an image uploader.

Good idea!

>  if we're using the dropzone on the sidebar to upload, why is this file input field necessary? Is it possible to trigger the dropzone programmatically?

So we actually are sending the files programatically to the dropzone. The difference is about events. If we just trigger the file upload on the dropzone and get the upload select box there's no way to identify what toolbar it came from.
Using an input box associated to the toolbar, and then sending the file to the dropzone lets us middle-man the process and inject the data regarding *where* the upload needs to call back and inject syntax into.

Reply:
Changed to only append the uploaded if the image button is used.